File: rules-interpreter/src/main/java/com/github/iamrenny/ruleflow/evaluators/PropertyContextEvaluator.java
package io.github.iamrenny.ruleflow.evaluators;

import io.github.iamrenny.ruleflow.RuleFlowLanguageParser.PropertyContext;
import io.github.iamrenny.ruleflow.errors.PropertyNotFoundException;
import io.github.iamrenny.ruleflow.visitors.Visitor;

public class PropertyContextEvaluator implements ContextEvaluator<PropertyContext> {

    @Override
    public Object evaluate(PropertyContext ctx, Visitor visitor) throws PropertyNotFoundException {
        ValidPropertyContextEvaluator validPropertyCondition = new ValidPropertyContextEvaluator();
        return validPropertyCondition.evaluate(ctx.validProperty(), visitor);
    }
}

File: rules-interpreter/src/main/kotlin/com/rappi/analang/evaluators/PropertyContextEvaluator.kt
package com.github.iamrenny.rulesflow.evaluators

import com.rappi.fraud.analang.ANAParser
import com.github.iamrenny.rulesflow.visitors.Visitor

class PropertyContextEvaluator: ContextEvaluator<ANAParser.PropertyContext> {
    override fun evaluate(ctx: ANAParser.PropertyContext, visitor: Visitor): Any {
        val validPropertyCondition = ValidPropertyContextEvaluator()
        return validPropertyCondition.evaluate(ctx.validProperty(), visitor)
    }
}
